ampd/src/queue/queued_broadcaster.rs

```diff
@@ -78,113 +79,99 @@ where
             queue: MsgQueue::default(),
             batch_gas_limit,
             broadcast_interval,
-            channel: mpsc::channel(capacity),
+            channel: Some(mpsc::channel(capacity)),
         }
     }
 
-    pub async fn run(self) -> Result {
-        let (tx, mut rx) = self.channel;
-        drop(tx);
-
-        let mut queue = self.queue;
-        let mut broadcaster = self.broadcaster;
-
+    pub async fn run(mut self) -> Result {
+        let (_, mut rx) = self
+            .channel
+            .take()
+            .expect("channel must be set before running");
```

Comment:
I would prefer an explanation why this shouldn't fail instead of a "regular" error message. Example:
"broadcast channel is expected to be set during initialization and must be available when running the broadcaster"